calyx-opt/src/passes/static_promotion.rs

```diff
@@ -190,6 +201,54 @@ impl Named for StaticPromotion {
 }
 
 impl StaticPromotion {
+    // Looks through ctx to get the given command line threshold.
+    // Default threshold = 1
+    fn get_threshold(ctx: &ir::Context) -> (u64, bool)
+    where
+        Self: Named,
+    {
+        let n = Self::name();
+
+        // getting the given opts for -x cell-share:__
+        let given_opts: HashSet<_> = ctx
```

Comment:
`get_opts` doesn't work here. 
It only works for arguments of the form `-x pass_name:bool_flag`. However, I want to do something like `-x static-promotion:threshold=4` which is slightly different, since threshold can take multiple values. `cell-share` has a similar problem, so eventually could probably have factored out code.

Reply:
Got it! It might be cool to support more complex argument parsing one day in the future but okay to use ad-hoc stuff for now!